prime-router/src/main/kotlin/serializers/Hl7Serializer.kt

```diff
@@ -41,6 +41,13 @@ class Hl7Serializer(val metadata: Metadata) {
     private val buildVersion: String
     private val buildDate: String
     private val formatter = DateTimeFormatter.ofPattern("yyyyMMddHHmmss.SSSSZZZ")
+    private var hl7Config: Hl7Configuration? = null
+
+    private val hdFieldMaximumLength: Int? get() = if (hl7Config?.truncateHDNamespaceIds == true) {
+        HD_TRUNCATION_LIMIT
+    } else {
+        null
+    }
 
```

Comment:
Added a property for the `hdFieldMaximumLength` based on whether or not we have an hl7Config passed in. This obviously means you better set the private var hl7Config before you start calling or you might not get the value you expect.

Reply:
FYI, I added a `maxLength: ` field to element types.    Currently its only used on output - that is, when taking an internal Report/Element in-memory data structure and turning it into a file.   (The initial use case was were were outputting a concat(field1, field2) that was too big.

Its currently only used in AZ schema, I think.     

However, its a simple concept, and could be enforced at time of ingest validation, even before we generate CSVs or HL7s.

Reply:
That only works if we have and use custom translations and schemas for each receiver. Going with a single translation type of HL7 that everyone consumes means I'd truncate for all or none, and some states don't want the HD namespace ID truncated (or don't care).